docs/data/date-pickers/shortcuts/EventImportance.tsx

```diff
@@ -0,0 +1,111 @@
+import dayjs, { Dayjs } from 'dayjs';
+import Stack from '@mui/material/Stack';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { PickerChangeEventImportance } from '@mui/x-date-pickers/models';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { PickersShortcutsItem } from '@mui/x-date-pickers/PickersShortcuts';
+import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
+import ToggleButton from '@mui/material/ToggleButton';
+
+const getMonthWeekday = (
+  monthIndex: number,
+  weekdayIndex: number,
+  dayRank: number,
+) => {
+  // Helper to find for example the 3rd monday in Jun
+  const today = dayjs();
+  const firstDayOfMonth = today.month(monthIndex).startOf('month');
+  const weekDay = firstDayOfMonth.day(); // 0 (Sunday) to 6 (Saturday)
+
+  const deltaToFirstValidWeekDayInMonth =
+    (weekDay > weekdayIndex ? 7 : 0) + weekdayIndex - weekDay;
+  return firstDayOfMonth.add(
+    (dayRank - 1) * 7 + deltaToFirstValidWeekDayInMonth,
+    'day',
+  );
+};
+
+const shortcutsItems: PickersShortcutsItem<Dayjs | null>[] = [
+  {
+    label: "New Year's Day",
+    getValue: () => {
+      // (January 1)
+
+      const today = dayjs();
+      return today.month(0).date(1);
+    },
+  },
+  {
+    label: 'Birthday of MLK Jr.',
+    getValue: () => {
+      // (third Monday in January)
+      return getMonthWeekday(0, 1, 3);
+    },
+  },
+  {
+    label: 'Independence Day',
+    getValue: () => {
+      // (July 4)
+
+      const today = dayjs();
+      return today.month(6).date(4);
+    },
+  },
+  {
+    label: 'Labor Day',
+    getValue: () => {
+      // (first Monday in September)
+      return getMonthWeekday(8, 1, 1);
+    },
+  },
+  {
+    label: 'Thanksgiving Day',
+    getValue: () => {
+      // (fourth Thursday in November)
+      return getMonthWeekday(10, 4, 4);
+    },
+  },
+  {
+    label: 'Christmas Day',
+    getValue: () => {
+      // (December 25)
+
+      const today = dayjs();
+      return today.month(11).date(25);
+    },
+  },
+];
+
+export default function EventImportance() {
+  const [eventImportance, setEventImportance] =
+    React.useState<PickerChangeEventImportance>('finish');
+
+  return (
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <Stack spacing={3} sx={{ width: 300 }}>
+        <ToggleButtonGroup
+          value={eventImportance}
+          exclusive
+          fullWidth
+          onChange={(event, newEventImportance) =>
+            setEventImportance(newEventImportance)
+          }
+        >
+          <ToggleButton value="finish">finish</ToggleButton>
+          <ToggleButton value="draft">draft</ToggleButton>
+          <ToggleButton value="shallow">shallow</ToggleButton>
+        </ToggleButtonGroup>
+        <DatePicker
+          slotProps={{
+            shortcuts: {
+              items: shortcutsItems,
+              eventImportance,
```

Comment:
Just Wondering: is there a specific reason why we call it `eventImportance`? 🤔 Somehow the naming is throwing me off because it seems that it actually impacts the behavior - but it may be that I am missing some context on this

Reply:
> Somehow the naming is throwing me off because it seems that it actually impacts the behavior

It does impact the behavior, doesn't it? 🤔 
I'm just thunking that maybe the name itself could be a bit different:
- `changeImportance`
- `selectImportance`
- `selectionImportance`

Reply:
I try to avoid `selection` because it can be confusing with the actual selection event on the input.
But I agree that it seems nice :laughing: 

`changeImportance` is fine for me

Reply:
Renamed and documented

I would also like to discuss the values of this property
`shallow` / `partial` / `finish` don't feel super understandable to me, but I struggle to find a good alternative.

Reply:
Indeed, these values make sense in the context of `selectionState` variable, but when we are talking about `changeImportance`, they seem out of context... 🤔 
Maybe going for `changeType` as the prop name would make more sense. 🙈 
If we'd go with `changeType`, the existing options do seem correct in that case. 🤷 

I do also struggle coming up with good options in case of `changeImportance`. Maybe that's the sign that the variable name is not the best for this particular case? 🙈 
What do you think about the following alternatives if we go with `changeImportance`?
- `shallow`: preselect
- `partial`: set
- `finish`: 🤷

Reply:
Yes, aligning those two concepts to use the same naming might not be the best approach. 🤔 

Another option to consider:
- `shallow` -> `set`
- `partial` -> `commit`
- `finish` -> `publish`

Reply:
We use `commit` for actions that calls `onAccept`, so `commit` for `partial` would not be consistent.
(this current usage of `commit` can be renamed if needs be though, it's 100% internal).

Reply:
I love adding new naming proposals when there are already too many options 😇

- `eventImportance` -> `action` or `shortcutAction`
- `shallow` -> `show` (if it does not display the value as selected; Otherwise I don't see the interest of this option)
- `partial` -> `set`
- `finish` -> `accept`

And an extra parameter `view` to call `onViewChange` with it.

Reply:
> I don't understand this part

Me too 🙈

---

I was probably thinking about our discussion n should shortcuts stay in the current view or move to the `openTo`, or something else.

My idea was to add a `view` property to shortcuts items, such that when you click on the shortcut, the picker moves to the expected view.

For example this item would set the date to the next year and move to year view to ensure the user noticed it
```js
{
    label: 'next year',
    getValue: () => { ... },
	view: 'year'
}
```

But that's probably another topic for another PR